Research/ResearchUI/iOS/Step View Controllers/RSDScrollingOverviewStepViewController.swift

```diff
@@ -83,11 +92,41 @@ open class RSDScrollingOverviewStepViewController: RSDOverviewStepViewController
             icon.image = nil
         }
         
-        if let icons = (self.step as? RSDOverviewStep)?.icons {
-            for (idx, iconInfo) in icons.enumerated() {
-                iconImages[idx].image = iconInfo.icon?.embeddedImage()
-                iconTitles[idx].text = iconInfo.title
+        if let overviewStep = self.step as? RSDOverviewStep {
+            
+            if let icons = overviewStep.icons {
+                
+                for (idx, iconInfo) in icons.enumerated() {
+                    iconImages[idx].image = iconInfo.icon?.embeddedImage()
+                    iconTitles[idx].text = iconInfo.title
+                }
+                
+                // TODO: syoung 03/12/2019 Change to using a collection view.
+                // When there are only 2 icons, employ this hack to center them evenly
+                if (icons.count == 2) {
+                    let removeIdx = 2
+                    
+                    // Adjust margin factor to give smaller screens more room
+                    let cellWidth = self.view.frame.size.width / CGFloat(iconImages.count)
+                    let marginFactor: CGFloat = (cellWidth < 125) ? 3.0 : 2.0
+                    
+                    // First, Adjust the leading/trailing spacing
+                    iconImagesLeadingConstraint.constant = cellWidth / marginFactor
+                    iconImagesTrailingConstraint.constant = cellWidth / marginFactor
+                    
+                    // Then, remove the third icon from the stack view
+                    iconImages[removeIdx].superview?.removeFromSuperview()
+                }
+            }
+
+            // Hide learn more action if it is not provided by the step json
+            if (self.step as? RSDOverviewStepObject)?.action(for: .navigation(.learnMore), on: self.step) == nil {
```

Comment:
So it turns out that upper-right icon wasn't using learn more, it was using the "reviewInstructions" navigation step, so I was able to just use the learn more properly  👍

Reply:
Looking at the nib in master, the info button is mapped to learnMore, but I agree that it makes more sense to map it to review instructions. That said, all of the "standard" navigation actions have a method in the base step view controller that map to the action.